src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,155 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int DEFAULT_SIZE = 0;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (size == DEFAULT_SIZE) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < DEFAULT_SIZE || index > size) {
+            throw new IndexOutOfBoundsException("IndexOutOfBoundsException: index "
+                    + index + " with size " + size);
+        }
+        if (size == index) {
+            add(value);
+        } else {
```

Comment:
```suggestion
        if (size == index) {
            add(value);
            return;
        }
```
lets not overcomplicate if else constructs